prometheus/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -97,6 +97,28 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
     [queryHandlerSettings, handleQueryChange]
   );
 
+  const handleSpecChange = (
```

Comment:
couldnt this one be called directly instead of having this chain of handleSpecChange -> handleOtherSpecsChanges ?